pkg/validate/dynamic/dynamic.go

```diff
@@ -576,10 +577,25 @@ func (dv *dynamic) validateCIDRRanges(ctx context.Context, subnets []Subnet, add
 	}
 
 	for _, c := range additionalCIDRs {
-		_, cidr, err := net.ParseCIDR(c)
+		IP, cidr, err := net.ParseCIDR(c)
 		if err != nil {
 			return err
 		}
+
+		if IP.IsUnspecified() {
+			return fmt.Errorf("address must be specified")
+		}
+
+		nip := IP.Mask(cidr.Mask)
+		if nip.String() != IP.String() {
+			return api.NewCloudError(
+				http.StatusBadRequest,
+				api.CloudErrorCodeInvalidNetworkAddress,
+				"",
+				errMsgInvalidCIDR,
+				err,
```

Comment:
Your error message contains 2 formatting directives (%s and %s), but I only see one string passed in here (err).

Also, I don't think that the var `err` will be set here, since you are just doing a comparison. Your error would say something like `Invalid network address got "", expecting %s.`

Maybe we need something like this?
```
	return api.NewCloudError(
				http.StatusBadRequest,
				api.CloudErrorCodeInvalidNetworkAddress,
				"",
				errMsgInvalidCIDR,
				nip.String(),
                                IP.String()
```

Reply:
Done